### PR TITLE
Add the embedded RPi Imager and document the list of available programs

### PR DESCRIPTION
**Pull request by timg236**
Also add some minor Readme.md tidyups and add a comment indicating that `initramfs` is likely to be required for boot security.

### CODE REVIEW

**Comment by timg236** [comment]
@peterharperuk

File: Readme.md [hunk 1]
```diff
@@ -1,24 +1,31 @@
...
 
 This is the USB MSD boot code which should work on the Raspberry Pi model A, Compute Module, Compute
 Module 3, Compute Module 4 and Raspberry Pi Zero.
 
-This version of rpiboot has been modified to work from directories which contain the booting
-firmware.  There is a msd/ directory which contains bootcode.bin and start.elf to turn
-the Raspberry Pi device into a USB Mass Storage Device (MSD). If run without arguments
-embedded versions of bootcode.bin and start.elf are used to enable the MSD behaviour.
+The default behaviour when run with no arguments is to boot the Raspberry Pi with
+special firmware sot that it emulates USB Mass Storage Device (MSD). The host OS
```
Comment (peterharperuk) [inline Readme.md:7]: "sot" => "so"

File: Readme.md [hunk 2]
```diff
@@ -1,24 +1,31 @@
...
+will treat this as a normal USB mass storage device and allowing the file-system
```
Comment (peterharperuk) [inline Readme.md:8]: remove word "and"

File: Readme.md [hunk 3]
```diff
@@ -1,24 +1,31 @@
...
+to be accessed. If the storage has not been formatted then (default for Compute Module)
```
Comment (pelwell) [inline Readme.md:9]: Delete 'then', or replace with 'yet'?

File: Readme.md [hunk 4]
```diff
@@ -1,24 +1,31 @@
...
+
+For more information run `rpiboot -h`
 
 ## Building
 
-### Ubuntu
-Clone this on your Pi or an Ubuntu linux machine
+### Linux / Cygwin / WSL
+Clone this on your Pi or a Linux machine.  
+Make sure that the system date is set correctly, otherwise, Git may produce an error.
```
Comment (pelwell) [inline Readme.md:24]: Second comma unnecessary.

File: Readme.md [hunk 5]
```diff
@@ -1,24 +1,31 @@
...
 
 ```
+sudo apt install git libusb-1.0-0-dev
 git clone --depth=1 https://github.com/raspberrypi/usbboot
-cd usbboot
```
Comment (pelwell) [inline Readme.md:20]: I think you still need the `cd`.

File: Readme.md [hunk 6]
```diff
@@ -104,11 +122,25 @@ package supports secure boot. To download the firmware files directly.
...
+in an [initramfs](https://www.raspberrypi.com/documentation/computers/config_txt.html#initramfs)
+and built with (buildroot)[https://buildroot.org/] or (yocto)[https://www.yoctoproject.org/].
+
+This ensures that all of the kernel modules and device tree dependencies are covered
+by the secure-boot signature.
+
+Since the `initramfs` is part of the `boot.img` it is possible to replace GPU firmware,
+kernel and dependencies in a single file update.
+
+Alternatively, for test/development the following instructions how a normal
```
Comment (pelwell) [inline Readme.md:136]: "following instructions _explain_ how"?

File: msd/README.md [hunk 7]
```diff
@@ -0,0 +1,10 @@
+This directory contains the second stage bootloader and firmware for
+booting the Raspberry Pi as a mass-storage device. The files here are
+embedded into the `rpiboot` executable as part of the build process.
+
+To load the files from this directory directly run.
```
Comment (pelwell) [inline msd/README.md:5]: "directory directly, run:"

File: rpi-imager-embedded/README.md [hunk 8]
```diff
@@ -0,0 +1,19 @@
+This directory contains the embedded Raspberry Pi Imager application. 
+Running RPi Imager on the target device is probably the simplest way
+to install the operating system on a Compute Module 4. However, it 
+does not export the CM4 file-system as a mass-storage device.
+
+The download the latest version run
```
Comment (pelwell) [inline rpi-imager-embedded/README.md:6]: "_To_ download the latest version, run:"

**Comment by timg236** [comment]
Also updated the installer packages, and fixed a long standing bug in windows installer where the .git directory would get included.

**Pull request merged**